### PR TITLE
internal/ethapi: add more testcase for block/header rpc

### PR DESCRIPTION
**Pull request by jsvisa**

### CODE REVIEW

File: internal/ethapi/api_test.go [hunk 1]
```diff
@@ -72,117 +73,140 @@ func TestTransaction_RoundTripRpcJSON(t *testing.T) {
...
+				GasPrice: big.NewInt(6),
+				Gas:      7,
+				To:       &addr,
+				Value:    big.NewInt(8),
+				Data:     []byte{0, 1, 2, 3, 4},
+				V:        big.NewInt(9),
+				R:        big.NewInt(10),
+				S:        big.NewInt(11),
+			},
+			Bs: `{"blockHash":null,"blockNumber":null,"from":"0x71562b71999873db5b286df957af199ec94617f7","gas":"0x7","gasPrice":"0x6","hash":"0x5f3240454cd09a5d8b1c5d651eefae7a339262875bcd2d0e6676f3d989967008","input":"0x0001020304","nonce":"0x5","to":"0xdead000000000000000000000000000000000000","transactionIndex":null,"value":"0x8","type":"0x0","chainId":"0x539","v":"0xa96","r":"0xbc85e96592b95f7160825d837abb407f009df9ebe8f1b9158a4b8dd093377f75","s":"0x1b55ea3af5574c536967b039ba6999ef6c89cf22fc04bcb296e0e8b0b9b576f5"}`,
```
Comment (s1na) [inline internal/ethapi/api_test.go:110]: Nit: it's not clear to me what Bs is short for. Please use `Want`.

Comment (jsvisa) [reply internal/ethapi/api_test.go:110]: Sorry, just want to indent with `Tx` :)

**Comment by s1na** [comment]
It's great you're adding more tests! while you're at it can you also cover the `ByHash` variants for header and block?

**Comment by jsvisa** [comment]
Done, please take another look

> It's great you're adding more tests! while you're at it can you also cover the `ByHash` variants for header and block?

**Comment by jsvisa** [comment]
Some of them are just for testing different boundaries and do not include the `Safe`, `finalized` block

After we merge this one, then I'm confident that I can complete the https://github.com/ethereum/go-ethereum/pull/27305.

**Comment by s1na** [comment]
Weird, one of the tests seems to be failing on windows: https://ci.appveyor.com/project/ethereum/go-ethereum/builds/47119845/job/owkbk2mxf3aygp5r?fullLog=true#L839

**Comment by jsvisa** [comment]
Size is not the same


```
want (string) (len=4) "size": (string) (len=5) "0x23e",
have (string) (len=4) "size": (string) (len=5) "0x21a",
```

diff is `36`

maybe related to 32bit?

I'm going to find a Windows PC to test it manually.

**Comment by jsvisa** [comment]
Seems is the reason of `headerSize` in 64 and 32 bits, check https://github.com/ethereum/go-ethereum/pull/27345 test case

Failed to run in win32 
image

**Comment by s1na** [comment]
Judging from `Header.Size` comment:

> It is used to approximate and limit the memory consumption of various caches.

It shouldn't be returned from API.

**Pull request merged**